modin/pandas/base.py

```diff
@@ -11,7 +11,34 @@
 # ANY KIND, either express or implied. See the License for the specific language
 # governing permissions and limitations under the License.
 
-"""Implement DataFrame/Series public API as pandas does."""
+"""
+Implement DataFrame/Series public API as pandas does.
+
+Methods docstrings contain tables with supported parameters information.
+These tables are structed as follows: The first column contains the parameter name,
+the second, third and fourth column - flags describing particular properties of
+method parameters for a concrete execution supported by Modin (`PandasOnRay`, `PandasOnDask`
+or `OmniSci`), and the last one is for parameter notes. The flags stand for the following:
+
++-----------+-----------------------------------------------------------------------------------------------+
+| Flag      | Meaning                                                                                       |
++===========+===============================================================================================+
+| Harmful   | Usage of this parameter can be harmfull for performance of your application                   |
++-----------+-----------------------------------------------------------------------------------------------+
+| Non-lazy  | Usage of this parameter can trigger non-lazy execution (actual for OmniSci execution only)    |
++-----------+-----------------------------------------------------------------------------------------------+
+| Partial   | Parameter can be partly unsupported, it's usage can be harmfull for performance of your       |
```

Comment:
harmful not harmfu**ll** here and in other places?

Reply:
Maybe we should even go from full words just to their first letters? Like "Harmful -> H", "Non-lazy -> NL", ... We may do so only for doc-strings and then replace the acronyms to the actual words when rendering the doc. IMO shorter the flag - less time to write and less chances to make a typo.

@alexander3774 what do you think? Wouldn't it be too hard to rename all entries of full words?

Reply:
> harmful not harmfu**ll** here and in other places?

According to the [Campbridge Dictionary](https://dictionary.cambridge.org/dictionary/essential-british-english/harmful) we should leave single l.

Reply:
> Maybe we should even go from full words just to their first letters? Like "Harmful -> H", "Non-lazy -> NL", ... We may do so only for doc-strings and then replace the acronyms to the actual words when rendering the doc. IMO shorter the flag - less time to write and less chances to make a typo.
> 
> @alexander3774 what do you think? Wouldn't it be too hard to rename all entries of full words?

@dchigarev it is not hard to replace flags with acronyms for summary supported APIs [table](https://modin--4286.org.readthedocs.build/en/4286/supported_apis/io_supported.html) in docs, but it can be tricky to modify generated [methods docs](https://modin--4286.org.readthedocs.build/en/4286/flow/modin/pandas/api/modin.pandas.read_csv.html), that are used as method full description, so i would suggest to leave full flags.

Reply:
We might just put some clear description of matching acronyms to real meaning in a header section of doc pages that contain a spreadsheet. And hence use acronyms in those spreadsheets

Reply:
Yeah, standalone doc for a function looks weird in this case. But it will require some explanation anyway, right? I mean if user came on this page NOT from supported APIs section, "harmful" would anyway require some explanation

Reply:
I think you are right, just stand alone `harmful` (or some another flag) is not anough, link to the the flags description table should be added in the notes section of the docstrings (this should be done in the #4286 because this PR doesn't contain flags description table).
Regarding acronyms - i would suggest to leave full words - it's really more error prone, but full words usage, as for me, makes table understanding more confinient.

Reply:
Agree

Reply:
Ok, "full" flags were added.